tests/ethereum/EVM/test_EVMSHIFT.py

```diff
@@ -0,0 +1,885 @@
+import struct
+import unittest
+import json
+from manticore.platforms import evm
+from manticore.core import state
+from manticore.core.smtlib import Operators, ConstraintSet
+import os
+
+
+class EVMTest_SHIFT(unittest.TestCase):
+    _multiprocess_can_split_ = True
+    maxDiff = None
+
+    def _execute(self, new_vm):
+        last_returned = None
+        last_exception = None
+        try:
+            new_vm.execute()
+        except evm.Stop as e:
+            last_exception = "STOP"
+        except evm.NotEnoughGas:
+            last_exception = "OOG"
+        except evm.StackUnderflow:
+            last_exception = "INSUFFICIENT STACK"
+        except evm.InvalidOpcode:
+            last_exception = "INVALID"
+        except evm.SelfDestruct:
+            last_exception = "SUICIDED"
+        except evm.Return as e:
+            last_exception = "RETURN"
+            last_returned = e.data
+        except evm.Revert:
+            last_exception = "REVERT"
+
+        return last_exception, last_returned
+
+    def test_SHL_0(self):
+        # Make the constraint store
+        constraints = ConstraintSet()
+        # make the ethereum world state
+        world = evm.EVMWorld(constraints)
+
+        address = 0x222222222222222222222222222222222222200
+        caller = 0x111111111111111111111111111111111111100
+        value = 10000
+        bytecode = b"\x1b"
+        data = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+        gas = 1000000
+
+        new_vm = evm.EVM(constraints, address, data, caller, value, bytecode, gas=gas, world=world)
+        new_vm._push(1)
+        new_vm._push(0)
+        last_exception, last_returned = self._execute(new_vm)
+        self.assertEqual(last_exception, None)
+        self.assertEqual(new_vm.pc, 1)
+        self.assertEqual(new_vm.stack, [1])
+
+    def test_SHL_1(self):
+        # Make the constraint store
+        constraints = ConstraintSet()
+        # make the ethereum world state
+        world = evm.EVMWorld(constraints)
+
+        address = 0x222222222222222222222222222222222222200
+        caller = 0x111111111111111111111111111111111111100
+        value = 10000
+        bytecode = b"\x1b"
+        data = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+        gas = 1000000
+
+        new_vm = evm.EVM(constraints, address, data, caller, value, bytecode, gas=gas, world=world)
```

Comment:
Eventually... we'll need a way to specify which fork are we testing. This opcode means a different thing in previous forks. Yes, ETH changed the meaning of the words that are used to write contracts. Smart.